app/src/main/java/uk/gov/onelogin/appinfo/service/data/AppInfoServiceImpl.kt

```diff
@@ -13,15 +14,23 @@ import uk.gov.onelogin.appinfo.source.domain.source.AppInfoRemoteSource
 
 class AppInfoServiceImpl @Inject constructor(
     private val remoteSource: AppInfoRemoteSource,
-    private val localSource: AppInfoLocalSource
+    private val localSource: AppInfoLocalSource,
+    private val appVersionCheck: AppVersionCheck
 ) : AppInfoService {
     override suspend fun get(): AppInfoServiceState {
         val localState = localSource.get()
         return when (val remoteResult = remoteSource.get()) {
             is AppInfoRemoteState.Success -> {
-                val encodedValue = Json.encodeToString<AppInfoData>(remoteResult.value)
-                this.localSource.update(encodedValue)
-                return AppInfoServiceState.RemoteSuccess(remoteResult.value)
+                // Check remote min version compatible
+                when (val compareResult = appVersionCheck.compareVersions(remoteResult.value)) {
+                    is AppInfoServiceState.Successful -> {
+                        // Encode value to save in SharedPrefs and save/ update
+                        val encodedValue = Json.encodeToString<AppInfoData>(remoteResult.value)
+                        localSource.update(encodedValue)
```

Comment:
This section in general; I wonder if we want to still save the app info locally even if the app needs updating 🤷  then this bit could look something like:
```
localSource.update(remoteResult.value)
return appVersionCheck.compareVersions(remoteResult.value)
```